server/broadcast.go

```diff
@@ -276,16 +277,25 @@ func processSegment(cxn *rtmpConnection, seg *stream.HLSSegment) error {
 
 	for {
 		// if fails, retry; rudimentary
-		if err := transcodeSegment(cxn, seg, name); err == nil {
+		err := transcodeSegment(cxn, seg, name)
+		// no error, transcode succeeded
+		if err == nil {
 			return nil
 		}
+
+		// non retryable error, close stream
```

Comment:
At the moment, we only close the stream for certain non-retryable errors. We don't close the stream for other non-retryable errors (i.e. "No sessions available"). I'd be ok with omitting the comment here entirely since the name of `shouldStopStream()` seems self-explanatory.